cpp/tests/table/experimental_row_operator_tests.cu

```diff
@@ -0,0 +1,300 @@
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
+
+  row_comparison(input_table_1,
+                 input_table_2,
+                 got->mutable_view(),
+                 column_order,
+                 lexicographic::sorting_physical_element_comparator{});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
+}
+
+TYPED_TEST(TypedTableViewTest, TestLexicographicalComparatorSameTable)
+{
+  using T = TypeParam;
+
+  fixed_width_column_wrapper<T> col1{{1, 2, 3, 4}};
+  std::vector<cudf::order> column_order{cudf::order::DESCENDING};
+
+  cudf::table_view input_table_1{{col1}};
+
+  auto got = cudf::make_numeric_column(
+    cudf::data_type(cudf::type_id::INT8), input_table_1.num_rows(), cudf::mask_state::UNALLOCATED);
+  fixed_width_column_wrapper<int8_t> expected{{0, 0, 0, 0}};
+
+  self_comparison(
+    input_table_1, got->mutable_view(), column_order, lexicographic::physical_element_comparator{});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
+
+  self_comparison(input_table_1,
+                  got->mutable_view(),
+                  column_order,
+                  lexicographic::sorting_physical_element_comparator{});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
+}
+
+TYPED_TEST(TypedTableViewTest, Select)
+{
+  using T = TypeParam;
+
+  fixed_width_column_wrapper<int8_t> col1{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<T> col2{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<int32_t> col3{{4, 5, 6, 7}};
+  fixed_width_column_wrapper<T> col4{{4, 5, 6, 7}};
+  cudf::table_view t{{col1, col2, col3, col4}};
+
+  cudf::table_view selected = t.select({2, 3});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(t.column(2), selected.column(0));
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(t.column(3), selected.column(1));
+}
+
+TYPED_TEST(TypedTableViewTest, SelectOutOfBounds)
+{
+  using T = TypeParam;
+
+  fixed_width_column_wrapper<int8_t> col1{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<T> col2{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<int32_t> col3{{4, 5, 6, 7}};
+  fixed_width_column_wrapper<T> col4{{4, 5, 6, 7}};
+  cudf::table_view t{{col1, col2}};
+
+  EXPECT_THROW((void)t.select({2, 3, 4}), std::out_of_range);
+}
+
+TYPED_TEST(TypedTableViewTest, SelectNoColumns)
+{
+  using T = TypeParam;
+
+  fixed_width_column_wrapper<int8_t> col1{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<T> col2{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<int32_t> col3{{4, 5, 6, 7}};
+  fixed_width_column_wrapper<T> col4{{4, 5, 6, 7}};
+  cudf::table_view t{{col1, col2, col3, col4}};
+
+  cudf::table_view selected = t.select({});
+  EXPECT_EQ(selected.num_columns(), 0);
+}
+
+template <typename T>
+struct NaNTableViewTest : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_SUITE(NaNTableViewTest, FloatingPointTypes);
+
+TYPED_TEST(NaNTableViewTest, TestLexicographicalComparatorTwoTableNaNCase)
+{
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<T> col1{{T(NAN), T(NAN), T(1), T(1)}};
```

Comment:
[`NAN` is a macro from `<cmath>`](https://en.cppreference.com/w/cpp/numeric/math/NAN). That should be included in this file.